vllm/distributed/device_communicators/shm_broadcast.py

```diff
@@ -154,19 +153,21 @@ def acquire_write(self):
                 written_flag = metadata_buffer[0]
                 if written_flag and read_count != self.buffer.n_reader:
                     # this block is written and not read by all readers
-                    # try to write to the next block
-                    self.current_idx = (self.current_idx +
-                                        1) % self.buffer.max_chunks
-                    if self.current_idx == start_index:
-                        # no empty block found
-                        if time.time(
-                        ) - start_time > VLLM_RINGBUFFER_WARNING_INTERVAL * n_warning:  # noqa
-                            logger.warning(
-                                "No available block found in %s second. ",
-                                VLLM_RINGBUFFER_WARNING_INTERVAL)
-                            n_warning += 1
-                        # wait for a while (0.1 us)
-                        time.sleep(1e-7)
+                    # for writers, `self.current_idx` is the next block to write
+                    # if this block is not ready to write,
+                    # we need to wait until it is read by all readers
+
+                    # wait for a while (0.1 us)
+                    time.sleep(1e-7)
```

Comment:
can you explain why?

Reply:
explain "# we need to wait until it is read by all readers" or "# wait for a while (0.1 us)" ?

Reply:
wait a while

Reply:
this affects how fast we can `broadcast`. if we wait for too long here, the `broadcast` will be slow. if we wait for too short, the cpu might be too busy. `0.1us` is a value that i think is fast enough, while won't keep cpu too busy when the next block is not ready.

Reply:
for reference, previously we use gloo or nccl to broadcast, and the latency is around 100us for even very small objects: https://docs.google.com/spreadsheets/d/1c9xgR0fGvm6SROfk7vrjwOZdYnKQk9oOafWK4_KgOyo/edit?gid=593626425#gid=593626425

Reply:
That makes sense, but can we put that in as a comment?

Reply:
fixed in [a3b9cf7](https://github.com/vllm-project/vllm/pull/5801/commits/a3b9cf726236439a3d1ecc4d5cad3af11da571b1)